packages/flutter/lib/src/widgets/list_wheel_scroll_view.dart

```diff
@@ -0,0 +1,713 @@
+    _hasScrolled();
+  }
+
+  /// Perspective of the cylindrical projection. See doc in
+  /// [MatrixUtils.createCylindricalProjectionTransform].
+  double get perspective => _perspective;
+  double _perspective;
+  set perspective(double value) {
+    assert(value != null);
+    assert(
+      value <= 0.01,
+      'A perspective value too high will always be clipped in the z axis and unrenderable.\n'
+      'Choose a value lower than 0.01',
+    );
+    if (value == _perspective)
+      return;
+    _perspective = value;
+    _hasScrolled();
+  }
+
+  /// The size of the children along the main axis. Children [RenderBox]es will
+  /// be given the [BoxConstraints] to be this exact size.
+  double get itemExtent => _itemExtent;
+  double _itemExtent;
+  set itemExtent(double value) {
+    assert(value != null);
+    assert(value > 0.0);
+    if (value == _itemExtent)
+      return;
+    _itemExtent = value;
+    markNeedsLayout();
+  }
+
+  bool _clipToSize;
+  set clipToSize(bool value) {
+    assert(value != null);
+    if (value == _clipToSize)
+      return;
+    _clipToSize = value;
+    _hasScrolled();
+  }
+
+  bool _renderChildrenOutsideViewport;
+  set renderChildrenOutsideViewport(bool value) {
+    assert(value != null);
+    if (value == _renderChildrenOutsideViewport)
+      return;
+    _renderChildrenOutsideViewport = value;
+    _hasScrolled();
+  }
+
+  void _hasScrolled() {
+    markNeedsPaint();
+    markNeedsSemanticsUpdate();
+  }
+
+  @override
+  void setupParentData(RenderObject child) {
+    if (child.parentData is! _ListWheelParentData)
+      child.parentData = new _ListWheelParentData();
+  }
+
+  @override
+  void attach(PipelineOwner owner) {
+    super.attach(owner);
+    _offset.addListener(_hasScrolled);
+  }
+
+  @override
+  void detach() {
+    _offset.removeListener(_hasScrolled);
+    super.detach();
+  }
+
+  @override
+  bool get isRepaintBoundary => true;
+
+  /// Main axis length in the untransformed plane.
+  double get _viewportExtent {
+    assert(hasSize);
+    return size.height;
+  }
+
+  /// Main axis scroll extent in the **scrollable layout coordinates** that puts
+  /// the first item in the center.
+  double get _minScrollExtent {
+    assert(hasSize);
+    return 0.0;
+  }
+
+  /// Main axis scroll extent in the **scrollable layout coordinates** that puts
+  /// the last item in the center.
+  double get _maxScrollExtent {
+    assert(hasSize);
+    if (!(childCount > 0))
+      return 0.0;
+
+    return math.max(0.0, (childCount - 1) * _itemExtent);
+  }
+
+  /// Scroll extent distance in the untransformed plane between the center
+  /// position in the viewport and the top position in the viewport.
+  ///
+  /// It's also the distance in the untransformed plane that children's painting
+  /// is offset by wrt to those children's [BoxParentData.offset].
+  double get _topScrollMarginExtent {
+    assert(hasSize);
+    // Consider adding an alignment configurable.
```

Comment:
configurable is an adjective, not a noun :-)

Reply:
Done